src/WorkItemMigrator/WorkItemImport/WitClient/WitClientUtils.cs

```diff
@@ -207,7 +208,11 @@ public void EnsureDateFields(WiRevision rev, WorkItem wi)
                     rev.Fields.Add(new WiField() { ReferenceName = WiFieldReference.ChangedDate, Value = rev.Time.AddMilliseconds(1).ToString("o") });
                 }
                 else
-                    rev.Fields.Add(new WiField() { ReferenceName = WiFieldReference.ChangedDate, Value = rev.Time.ToString("o") });
+                {
+                    // Seen while testing: DevOps can add a few milliseconds to work item createdDate, hence adding more here to the revision time
+                    rev.Fields.Add(new WiField() { ReferenceName = WiFieldReference.ChangedDate, Value = rev.Time.AddMilliseconds(5).ToString("o") });
+                    wi.Fields[WiFieldReference.ChangedDate] = rev.Time.AddMilliseconds(5);
+                }
```

Comment:
Any more info on why ADO behaves like this? It would be nice to know before we merge the change

Reply:
@VincentVerweij Turned out to be an issue with how the DateTime is parsed. Trying to convert a DateTime by using it's default string representation apparently truncates the miliseconds, so there will be a diff. That's propbably why you experienced issues and had to add 5 miliseconds here and there.

I also fixed some more minor problems with the ChangedDate handling. See: https://github.com/solidify/jira-azuredevops-migrator/pull/648